covid_moonshot_ml/docking/analysis.py

```diff
@@ -0,0 +1,272 @@
+            }
+        )
+
+    def calculate_rmsd_and_posit_score(self, fragalysis_dir):
+        rmsds = []
+        posit_scores = []
+        docking_scores = []
+        print(self.df.head().to_dict(orient="index"))
+        for data in self.df.to_dict(orient="index").values():
+            cmpd_id = data["Compound_ID"]
+            sdf_fn = data["SDF_Filename"]
+            ref_fn = data["Reference"]
+
+            cmpd_dir = self.get_cmpd_dir_path(cmpd_id)
+
+            sdf_path = os.path.join(cmpd_dir, sdf_fn)
+            ref_path = os.path.join(fragalysis_dir, ref_fn)
+
+            print(f"Loading rmsd calc on {sdf_path} compared to {ref_path}")
+
+            docking_results = get_ligand_rmsd_from_pdb_and_sdf(
+                ref_path, mobile_path=sdf_path, fetch_docking_results=True
+            )
+            posit_scores.append(docking_results["posit"])
+            docking_scores.append(docking_results["chemgauss"])
+            rmsds.append(docking_results["rmsd"])
+
+        self.df["POSIT"] = posit_scores
+        self.df["Chemgauss4"] = docking_scores
+        self.df["RMSD"] = rmsds
+
+    def write_csv(self, output_csv_fn):
+        self.df.to_csv(output_csv_fn, index=False)
+
+    def analyze_docking_results(
+        self, fragalysis_dir, output_csv_fn, test=False
+    ):
+        self.organize_docking_results()
+        # self.to_df()
+        if test:
+            self.df = self.df.head()
+        self.calculate_rmsd_and_posit_score(fragalysis_dir)
+        self.write_csv(output_csv_fn=output_csv_fn)
+
+
+def get_good_score(score):
+    """
+    The idea here is that given an array from a DataFrame, x, that is a particular score type, there should be a
+    single function that will tell you whether we think this score is "good" or not. I'm not sure this is the best way
+    to do this but someway to make it obvious when we are using which cutoff would be good.
+
+    Parameters
+    ----------
+    score
+
+    Returns
+    -------
+
+    """
+    if score == "RMSD":
+        lambda_func = lambda x: x[(x <= 2.5)].count()
+    elif score == "POSIT":
+        lambda_func = lambda x: x[(x > 0.7)].count()
+    elif score == "Chemgauss4":
+        lambda_func = lambda x: x[(x < 0)].count()
+    elif score == "POSIT_R":
+        lambda_func = lambda x: x[(x < 0.3)].count()
+    else:
+        raise NotImplementedError(
+            f"good score acquisition not implemented for {score}"
+        )
+    return lambda_func
+
+
+class DockingResults:
+    """
+    This is a class to parse docking results from a csv file.
+    Mainly for mainipulating the data in various useful ways.
+    """
+
+    def __init__(self, csv_path):
+        ## load in data and replace the annoying `-1.0` and `-1` values with nans
+        self.df = (
+            pd.read_csv(csv_path).replace(-1.0, np.nan).replace(-1, np.nan)
+        )
+
+    def get_grouped_df(
+        self, groupby_ID_column="Compound_ID", score_columns=SCORE_COLUMNS
```

Comment:
needs docstring or more comments throughout to clarify process

Reply:
done here: https://github.com/choderalab/covid-moonshot-ml/pull/39/commits/d4f20d67092a548d1850a5d413a8c184a6ac9f64